src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/RightCurlyCheck.java

```diff
@@ -779,6 +806,43 @@ private static Details getDetailsForOthers(DetailAST ast) {
             return new Details(lcurly, rcurly, getNextToken(ast), true);
         }
 
+        /**
+         * Collects validation details for anonymomus inner class.
+         *
+         * @param ast a {@code DetailAST} value
+         * @return an object containing all details to make a validation
+         */
+        private static Details getDetailsForAnonymousInnerClass(DetailAST ast) {
+
+            DetailAST rcurly = null;
+            DetailAST lcurly = null;
+            DetailAST nextToken = null;
+
+            if(checkIfDeclarationHasAnonymousInnerClass(ast)){
+                DetailAST objBlock = ast.findFirstToken(TokenTypes.OBJBLOCK);
+                rcurly = objBlock.getFirstChild();
+                lcurly = objBlock.getLastChild();
+                nextToken = getNextToken(objBlock);
+            }
+            return new Details(lcurly, rcurly, nextToken, true);
+        }
+
+        private static boolean checkIfDeclarationHasAnonymousInnerClass(DetailAST ast){
+        boolean ans = false;
+
+        DetailAST childNode = ast.getLastChild().getPreviousSibling();
+
+        while(childNode != null){
+            if(childNode.getType() != TokenTypes.OBJBLOCK){
+                ans = true;
+                break;
+            }
+            childNode = childNode.getLastChild();
+        }
```

Comment:
Here I am checking if there is any ``OBJ_BLOCK`` within a ``VARIABLE_DEF`` to specifically find if it has a anonymous inner class or not. If it does, then ``ans = true`` and return that.